SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -1118,3 +1134,43 @@ void ASpatialFunctionalTest::ClearAllTakenSnapshots()
 	bWasLoadedFromTakenSnapshot = false;
 	TakenSnapshots.Empty();
 }
+
+void ASpatialFunctionalTest::GenerateMap()
+{
+	bIsGeneratingMap = true;
+	GeneratedTestMap->GenerateMap();
+	CreateCustomContentForMap();
+	GeneratedTestMap->AddActorToLevel(GeneratedTestMap->GetWorld()->GetCurrentLevel(), this->GetClass(), FTransform(TestPositionInWorld));
+	bIsGeneratingMap = false;
+}
+
+bool ASpatialFunctionalTest::ShouldGenerateMap()
+{
+	return bIsStandaloneTest;
+}
+
+bool ASpatialFunctionalTest::SaveMap()
+{
+	return GeneratedTestMap->SaveMap();
+}
+
+bool ASpatialFunctionalTest::GenerateCustomConfig()
+{
+	return GeneratedTestMap->GenerateCustomConfig();
+}
+
+FString ASpatialFunctionalTest::GetMapName()
+{
+	return GeneratedTestMap->GetMapName();
+}
+
+void ASpatialFunctionalTest::SetCustomConfig(FString& String)
+{
+	GeneratedTestMap->SetCustomConfig(String);
+}
+
+ASpatialWorldSettings* ASpatialFunctionalTest::GetWorldSettings()
+{
+	checkf(bIsGeneratingMap, TEXT("GetWorldSettings should only be called from within an overridden CreateCustomContentForMap."));
```

Comment:
Why this limitation?

Reply:
The world returned by generatedTestMap->GetWorld() only returns a valid world while the map is being generated; and not at test runtime. I appreciate this method name on spatialfunctionaltest could be misleading; I've renamed the method to `GetWorldSettingsForMap` along with `SetCustomConfig` to `SetCustomConfigForMap`